core/src/main/java/io/grpc/internal/DelayedClientTransport2.java

```diff
@@ -105,54 +144,63 @@ public final Runnable start(Listener listener) {
   @Override
   public final ClientStream newStream(MethodDescriptor<?, ?> method, Metadata headers,
       CallOptions callOptions, StatsTraceContext statsTraceCtx) {
-    SubchannelPicker picker = null;
-    synchronized (lock) {
-      if (!shutdown) {
-        if (lastPicker == null) {
-          return createPendingStream(method, headers, callOptions, statsTraceCtx);
+    try {
+      SubchannelPicker picker = null;
+      long pickerVersion = -1;
```

Comment:
No, they don't. I put all lines in `try` just to emphasize the idea that `drain()` must be called when this method returns.